rosdep/osx-homebrew.yaml

```diff
@@ -650,7 +650,7 @@ protobuf-dev:
 python:
   osx:
     homebrew:
-      packages: []
+      packages: [python, python3]
```

Comment:
Looking at https://docs.brew.sh/Homebrew-and-Python#python-3y and https://formulae.brew.sh/formula/python@3.11

I actually think we should make the `python` rosdep key, which is implicitly python2, undefined on homebrew, since python2 has been removed.

Reply:
Is there an argument to remove it then, or will it just provide users a warning?

Reply:
The change as proposed is incorrect as both the `python` and `python3` formula names will result in python3 being installed. By removing the definition, any package relying on `python` (implicitly python2, and no longer available on macOS) will see a failure to resolve the rosdep key. Which is the preferred behavior if a dependency is not resolvable.

Reply:
I pushed [3ae1423](https://github.com/ros/rosdistro/pull/36747/commits/3ae14238aa26b2c3796b169f90def664593a5088) implementing my intended suggestion. There is a difference in rosdep between an "undefined" definition and an "empty" definition. An empty package list means that installing no packages satisfies this key. It is used in situations where the packages listed are part of the core system and therefore not installed by any specific package and other special situations where a package manager may actually be responsible for managing a package. To make the error condition we want for this key on macOS, we have to either assign it a value of `null` or, as I've done here, remove the definition entirely since there are no other definitions for this key in this file.